eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -256,101 +164,113 @@ class Router(nodeParams: NodeParams, watcher: ActorRef) extends FSM[State, Data]
 
     case Event(c: ChannelAnnouncement, d) =>
       log.debug(s"received channel announcement for shortChannelId=${c.shortChannelId.toHexString} nodeId1=${c.nodeId1} nodeId2=${c.nodeId2} from $sender")
-      if (d.channels.containsKey(c.shortChannelId) || d.awaiting.keys.exists(_.shortChannelId == c.shortChannelId) || d.stash.channels.contains(c)) {
+      if (d.channels.contains(c.shortChannelId) || d.awaiting.contains(c)) {
+        sender ! TransportHandler.ReadAck(c)
         log.debug("ignoring {} (duplicate)", c)
         stay
       } else if (!Announcements.checkSigs(c)) {
+        sender ! TransportHandler.ReadAck(c)
```

Comment:
If Announcements.checkSigs(c) throws an exception, c will never be acked (router will resume and c will be lost)